src/Language/PureScript/Make.hs

```diff
@@ -297,6 +289,24 @@ makeIO f io = do
 readTextFile :: FilePath -> Make String
 readTextFile path = makeIO (const (ErrorMessage [] $ CannotReadFile path)) $ readUTF8File path
 
+-- | Infer the module name for a module by looking for the same filename with
+-- a .js extension.
+inferForeignModules
+  :: forall m
+   . MonadIO m
+  => M.Map ModuleName (Either RebuildPolicy FilePath)
+  -> m (M.Map ModuleName FilePath)
+inferForeignModules = fmap (M.mapMaybe id) . traverse inferForeignModule
+  where
+    inferForeignModule :: Either RebuildPolicy FilePath -> m (Maybe FilePath)
+    inferForeignModule (Left _) = return Nothing
+    inferForeignModule (Right path) = do
+      let jsFile = replaceExtension path "js"
```

Comment:
I thought the plan was `path ++ ".js"`? So, eg, you might have `src/Data/Show.purs.js`.